rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1661,6 +1661,77 @@ func Qualify_Shaping: <"The qualification of a shaping event from the fact that
 		and packageRef count = openTradeNoExecutionDetails count
 		and packageRef distinct count = 1
 
+
+func Create_SubstitutionPrimitiveInstruction: <"Creates the primitive instructions for a substitution of collateral by replacing the assetpayout of the trade.">
+	inputs:
+		//ICMA-P2 Instructions
+		tradeState TradeState (1..1) <"The original trade to be rolled.">
+		effectiveDate AdjustableOrRelativeDate (1..1) <"The date to close and open a new position.">
+		newPayout Payout(1..1) <"New payout to subtitute for the original payout. The payout define the product and other contractual attributes.">
+		priceQuantity PriceQuantity (1..*) <"The price and quantity of the substituted product. ">
```

Comment:
A few points on the inputs. The comment for tradeState mentions "rolled" which is ambiguous here as we are not rolling, we are substituting collateral. The comment for effectiveDate implies that a new position will be opened. For securities lending a collateral substitution does not require the original loan to be terminated and a new one opened. I agree that all these inputs should be mandatory.